tests/ui/consts/cycle-static-promoted.rs

```diff
@@ -0,0 +1,12 @@
+// check-pass
+
+struct Value {
+    values: &'static [&'static Value],
+}
+
+// This cycle goes through a promoted (the slice).
```

Comment:
There's no cycle here anymore.

Reply:
I'd say there's still a recursive cycle with the two items pointing to each other, just the queries are no longer cyclic.